src/content/language/constructors.md

```diff
@@ -402,6 +340,211 @@ var logMap = {'name': 'UI'};
 var loggerJson = Logger.fromJson(logMap);
 ```
 
+### Redirecting factory constructors
+
+A redirecting factory constructor specifies a call to a constructor of another
+class to use whenever someone makes a call to the redirecting constructor.
+
+```dart
+factory Listenable.merge(List<Listenable> listenables) = _MergingListenable
+```
+
+It might appear that ordinary factory constructors
+could create and return instances of other classes.
+This would make redirecting factories unnecessary.
+Redirecting factories have several advantages:
+
+* An abstract class might provide a constant constructor
+  that uses the constant constructor of another class.
+* A redirecting factory constructor avoids the need for forwarders
+  to repeat the formal parameters and their default values.
+
+## Parameter initialization
+
+Dart can initialize parameters in a constructor in three ways.
+
+### Initialize parameters when declaring variables
+
+Initialize the constructor parameters when you declare variables.
+
+<?code-excerpt "point_alt.dart (initialize-declaration)" plaster="none"?>
+```dart
+class PointA {
+  double x = 1.0;
+  double y = 2.0;
+
+  // The parameterless constructor is not even be needed to set to (1.0,2.0)
+  PointA();
+
+  @override
+  String toString() {
+    return 'PointA($x,$y)';
+  }
+}
+```
+
+### Use initializing formal parameters
+
+To simplify the common pattern of assigning a constructor argument
+to an instance variable, Dart has *initializing formal parameters*.
+
+In the constructor declaration, include `this.<propertyName>`
+and omit the body. The `this` keyword refers to the current instance.
+
+When the name conflict exists, use `this`.
+Otherwise, Dart style omits the `this`.
+An exception exists for the generative constructor where
+you must prefix the initializing formal parameter name with `this`.
+
+As noted earlier in this guide, certain constructors
+and certain parts of constructors can't access `this`. These include:
+
+* Factory constructors
+* The right-hand side of an initializer list
+* Arguments to a superclass constructor
+
+Initializing formal parameters also allow you to initialize
+non-nullable or `final` instance variables.
+Both of these types of variables require initialization or a default value.
+
+<?code-excerpt "point_alt.dart (initialize-formal)" plaster="none"?>
+```dart
+class PointB {
+  final double x;
+  final double y;
+
+  // Sets the x and y instance variables
+  // before the constructor body runs.
+  PointB(this.x, this.y);
+
+  // Initializing formal parameters can also be optional.
+  PointB.optional([this.x = 0.0, this.y = 0.0]);
+  PointB.named({required this.x, required this.y});
+}
```

Comment:
With the current layout of the doc the last line is covered well in `PointC`- I think it flows well now to omit the mention here. The optional positional parameters are still useful to show here.

```suggestion
  // Initializing formal parameters can also be optional.
  PointB.optional([this.x = 0.0, this.y = 0.0]);
}
```

Reply:
Fair, but then we should "name" `PointC` constructors to illustrate the point.